tests/tests/test_train_schedule.py

```diff
@@ -301,6 +304,213 @@ def test_etcs_schedule_result_stop_brake_from_mrsp(etcs_scenario: Scenario, etcs
+    _assert_equal_speeds(speed_at_start_release_speed, RELEASE_SPEED_40)
+    offset_end_release_speed = 40_892_792
+    speed_at_end_release_speed = _get_current_or_next_speed_at(simulation_final_output, offset_end_release_speed)
+    _assert_equal_speeds(speed_at_end_release_speed, RELEASE_SPEED_40)
+
+
+def test_etcs_schedule_result_stop_with_eoa_and_svl_at_different_locations(
+    etcs_scenario: Scenario, etcs_rolling_stock: int
+):
+    rolling_stock_response = requests.get(EDITOAST_URL + f"light_rolling_stock/{etcs_rolling_stock}")
+    etcs_rolling_stock_name = rolling_stock_response.json()["name"]
+    ts_response = requests.post(
+        f"{EDITOAST_URL}timetable/{etcs_scenario.timetable}/train_schedules/",
+        json=[
+            {
+                "train_name": "brake from MRSP: max_speed + EoA and SvL 100m after EoA",
+                "labels": [],
+                "rolling_stock_name": etcs_rolling_stock_name,
+                "start_time": "2024-01-01T07:00:00Z",
+                "path": [
+                    {"id": "zero", "track": "TA0", "offset": 862_000},
+                    {"id": "first", "track": "TH0", "offset": 900_000},
+                    {"id": "last", "track": "TH1", "offset": 3_922_000},
+                ],
+                "schedule": [
+                    {"at": "zero", "stop_for": "P0D"},
+                    {"at": "first", "stop_for": "PT10S"},
+                    {"at": "last", "stop_for": "P0D"},
+                ],
+                "margins": {"boundaries": [], "values": ["0%"]},
+                "initial_speed": 0,
+                "comfort": "STANDARD",
+                "constraint_distribution": "STANDARD",
+                "speed_limit_tag": "foo",
+                "power_restrictions": [],
+            }
+        ],
+    )
+
+    schedule = ts_response.json()[0]
+    schedule_id = schedule["id"]
+    ts_id_response = requests.get(f"{EDITOAST_URL}train_schedule/{schedule_id}/")
+    ts_id_response.raise_for_status()
+    simu_response = requests.get(
+        f"{EDITOAST_URL}train_schedule/{schedule_id}/simulation?infra_id={etcs_scenario.infra}"
+    )
+    simulation_final_output = simu_response.json()["final_output"]
+
+    assert len(simulation_final_output["positions"]) == len(simulation_final_output["speeds"])
+
+    # To debug this test: please add a breakpoint then use front to display speed-space chart
+    # (activate Context for Slopes and Speed limits).
+
+    # This case hits an LoA curve (slowdown of the MRSP), but it's not the point to test it here.
+
+    # Check that the curves does respect the EoA + SvL (EoA = stops), and that there is an
+    # acceleration then deceleration in between (maintain speed when reach the MRSP).
+    # Here, the EoA is at 41038m, and the SvL is at the switch at 41138m.
+    distance_to_svl = 100_000
+    first_stop_offset = SVL_STOP_OFFSET - distance_to_svl
```

Comment:
Following the same idea:
```suggestion
    # Check that the curves does respect the EoA + SvL (EoA = stops), and that there is an
    # acceleration then deceleration in between (maintain speed when reach the MRSP).
    # Here, stop (EoA) is 100m before the PH1 switch (SvL).
    svl_ph1_offset = 41_138_000
    first_stop_offset = svl_ph1_offset - 100_000
```